src/BudgetManager/BudgetManager.CQRS/Validators/SubCategoryValidators/UpdateSubCategoryValidator.cs

```diff
@@ -0,0 +1,35 @@
+﻿using BudgetManager.SDK.DTOs;
+using FluentValidation;
+using MediatR;
+using BudgetManager.CQRS.Queries.SubCategoryQueries;
+using BudgetManager.CQRS.Responses.SubCategoryResponses;
+
+namespace BudgetManager.CQRS.Validators.SubCategoryValidators
+{
+    public class UpdateSubCategoryValidator : AbstractValidator<UpdateSubCategoryDTO>
+    {
+        private IEnumerable<SubCategoryResponse> _subCategories;
+        private readonly IMediator _mediator;
+
+        public UpdateSubCategoryValidator(IMediator mediator)
+        {
+            RuleFor(x => x.Name).NotEmpty()
+                .Must(IsNameUnique).WithMessage($"SubCategory with this 'Name' already exists")
+                .MaximumLength(100);
+
+            RuleFor(x => x.Color).NotEmpty();
+
+            _mediator = mediator;
+        }
+
+        public async Task SetUserAndCategory(Guid userId, Guid categoryId, CancellationToken cancellationToken)
+        {
+            _subCategories = await _mediator.Send(new GetSubCategoriesQuery(userId, categoryId), cancellationToken);
+        }
+
+        public bool IsNameUnique(UpdateSubCategoryDTO subCategory, string newValue)
+        {
+            return _subCategories.All(s => s.Name.ToLower() != newValue.ToLower() || s.Id == subCategory.SubCategoryId);
+        }
```

Comment:
This should be changed. How it is right now, it will force the user to also change the name of subcategory if it is kept the same.

Reply:
I tested changing subcategory's 'color' while keeping 'name' the same and it works fine.

Reply:
You are right, I didn't read " || s.Id == subCategory.SubCategoryId" part